src/MultiRegion/multi_region_cubed_sphere_grid.jl

```diff
@@ -247,89 +248,88 @@ function ConformalCubedSphereGrid(arch::AbstractArchitecture=CPU(), FT=Float64;
-                # so we compute them via conformal_cubed_sphere_mapping
-                φc, λc = cartesian_to_lat_lon(conformal_cubed_sphere_mapping(1, -1)...)
-                getregion(grid, region).φᶠᶠᵃ[1, Ny+1] = φc
-                getregion(grid, region).λᶠᶠᵃ[1, Ny+1] = λc
-            elseif iseven(region)
-                # Coordinates of "missing" SE corner points on even panels can't be read from the interior
-                # so we compute them via conformal_cubed_sphere_mapping
-                φc, λc = -1 .* cartesian_to_lat_lon(conformal_cubed_sphere_mapping(-1, -1)...)
-                getregion(grid, region).φᶠᶠᵃ[Nx+1, 1] = φc
-                getregion(grid, region).λᶠᶠᵃ[Nx+1, 1] = λc
-            end
+    ###################################################
+    ## Code specific to one-region-per panel partitions
+    ###################################################
+
+    ## hardcoding NW/SE corner values only works for a one-region-per panel partition
+
+    number_of_regions(grid) !== 6 && error("requires cubed sphere grids with 1 region per panel")
+
+    for region in 1:number_of_regions(grid)
+        if isodd(region)
+            # Coordinates of "missing" NW corner points on odd panels can't be read from the interior
+            # so we compute them via conformal_cubed_sphere_mapping
+            φc, λc = cartesian_to_lat_lon(conformal_cubed_sphere_mapping(1, -1)...)
+            getregion(grid, region).φᶠᶠᵃ[1, Ny+1] = φc
+            getregion(grid, region).λᶠᶠᵃ[1, Ny+1] = λc
+        elseif iseven(region)
+            # Coordinates of "missing" SE corner points on even panels can't be read from the interior
+            # so we compute them via conformal_cubed_sphere_mapping
+            φc, λc = -1 .* cartesian_to_lat_lon(conformal_cubed_sphere_mapping(-1, -1)...)
+            getregion(grid, region).φᶠᶠᵃ[Nx+1, 1] = φc
+            getregion(grid, region).λᶠᶠᵃ[Nx+1, 1] = λc
         end
 
-        for region in 1:number_of_regions(grid)
-            getregion(grid, region).λᶜᶜᵃ[getregion(grid, region).λᶜᶜᵃ .== -180] .= 180
-            getregion(grid, region).λᶠᶜᵃ[getregion(grid, region).λᶠᶜᵃ .== -180] .= 180
-            getregion(grid, region).λᶜᶠᵃ[getregion(grid, region).λᶜᶠᵃ .== -180] .= 180
-            getregion(grid, region).λᶠᶠᵃ[getregion(grid, region).λᶠᶠᵃ .== -180] .= 180
-        end
-    end # CUDA.@allowscalar
+        getregion(grid, region).λᶜᶜᵃ[getregion(grid, region).λᶜᶜᵃ .== -180] .= 180
+        getregion(grid, region).λᶠᶜᵃ[getregion(grid, region).λᶠᶜᵃ .== -180] .= 180
+        getregion(grid, region).λᶜᶠᵃ[getregion(grid, region).λᶜᶠᵃ .== -180] .= 180
+        getregion(grid, region).λᶠᶠᵃ[getregion(grid, region).λᶠᶠᵃ .== -180] .= 180
+    end
+
+    # now convert to proper architecture
+    archs = tuple(fill(arch, number_of_regions(grid))...)
+
+    new_regional_objects = on_architecture.(archs, grid.region_grids.regional_objects)
```

Comment:
I am not sure there is a `on_architecture` method for a `MultiRegionObject` but we should add it so that you can directly write
```julia
new_region_grids = on_architecture(arch, region_grids)
```

Reply:
also, we should probably have an `on_architecture` method for a `MultiRegionGrid` directly, so we can avoid this and just write
```julia
return on_architecture(arch, grid)
```